src/libc/tolower.src

```diff
@@ -3,24 +3,20 @@
 	section	.text
 	public	_tolower
 
-if PREFER_OS_LIBC
-
-_tolower := 021E34h
-
-else
-
 _tolower:
 	pop	de
 	ex	(sp), hl
 	push	de
+	require	_tolower.internal
 
+	section	.text
+	public	_tolower.internal
+	; ASM interface: input/output char in L, destroys AF
+_tolower.internal:
 	ld	a, l
 	sub	a, 'A'
 	cp	a, 1+'Z'-'A'
 	ret	nc
 
-	add	a, 'a'
-	ld	l, a
```

Comment:
How does this work

Reply:
Lowercase and uppercase letters in ASCII differ only by bit 5. Incidentally, that's also how the optimized loops for str[n]casecmp work by doing an xor check between the two characters.